test/python3/libdnf/rpm/test_reldep_list.py

```diff
@@ -0,0 +1,158 @@
+
+        self.assertEqual(list1.get(0), a)
+        self.assertEqual(list1.get(1).to_string(), "(lab-list if labirinto.txt)")
+        self.assertEqual(list1.get_id(2).id, c.get_id().id)
+
+    def test_compare(self):
+        a = libdnf.rpm.Reldep(self.sack, "python3-labirinto = 4.2.0")
+        b = libdnf.rpm.Reldep(self.sack, "(lab-list if labirinto.txt)")
+
+        list1 = libdnf.rpm.ReldepList(self.sack)
+        list2 = libdnf.rpm.ReldepList(self.sack)
+        self.assertEqual(list1, list2)
+
+        list1.add(a)
+        self.assertNotEquals(list1, list2)
+
+        list2.add(a)
+        list1.add(b)
+        list2.add(b)
+        self.assertEqual(list1, list2)
+
+    def test_append(self):
+        a = libdnf.rpm.Reldep(self.sack, "python3-labirinto = 4.2.0")
+        b = libdnf.rpm.Reldep(self.sack, "(lab-list if labirinto.txt)")
+        c = libdnf.rpm.Reldep(self.sack, "(labirinto unless labirinto_c)")
+        d = libdnf.rpm.Reldep(self.sack, "labirinto.txt")
+
+        list1 = libdnf.rpm.ReldepList(self.sack)
+        list1.add(a)
+        list1.add(b)
+        list1.add_reldep("delgado > 1.2")
+
+        list2 = libdnf.rpm.ReldepList(self.sack)
+        list2.add(c)
+        list2.add(d)
+
+        list1.append(list2)
+        self.assertEqual(list1.size(), 5)
+        self.assertEqual(list1.get(0), a)
+        self.assertEqual(list1.get(1), b)
+        self.assertEqual(list1.get(2).to_string(), "delgado > 1.2")
+        self.assertEqual(list1.get(3), c)
+        self.assertEqual(list1.get(4), d)
+
+    def test_iterator(self):
+        a = libdnf.rpm.Reldep(self.sack, "python3-labirinto = 4.2.0")
+        b = libdnf.rpm.Reldep(self.sack, "(lab-list if labirinto.txt)")
+        c = libdnf.rpm.Reldep(self.sack, "(labirinto unless labirinto_c)")
+        d = libdnf.rpm.Reldep(self.sack, "labirinto.txt")
+        expect = [a, b, c, d]
+
+        list1 = libdnf.rpm.ReldepList(self.sack)
+        for reldep in expect:
+            list1.add(reldep)
+
+        # check if begin() points to the first Reldep
+        it1 = list1.begin()
+        self.assertEqual(it1.value(), a)
+
+        # test next()
+        it1.next()
+        self.assertEqual(it1.value(), b)
+
+        # check Python style iterator
+        pit = iter(list1)
+        self.assertEqual(next(pit), a)
+        self.assertEqual(next(pit), b)
+        self.assertEqual(next(pit, None), c)
+        self.assertEqual(next(pit, None), d)
+        self.assertEqual(next(pit, None), None)
+
+        # test iteration using a for loop
+        result = []
+        for reldep in list1:
+            result.append(reldep)
+        self.assertEqual(result, expect)
+
+    # add_reldep_with_glob uses libsolvs Dataiterator which needs the actual packages
+    def test_add_reldep_with_glob(self):
+        list1 = libdnf.rpm.ReldepList(self.sack)
+        list1.add_reldep_with_glob("dwm*")
+
+        expected = ["dwm-6.1-1.fc29.spec", "dwm", "dwm", "dwm(x86-64)", "dwm(x86-64)"]
```

Comment:
""dwm-6.1-1.fc29.spec"" looks like something incorrect.

Reply:
I don't know. All I know is that there is a package in the test repository
`dwm-6.1-1.src.rpm`, which contains the `dwm-6.1-1.fc29.spec` file.

and the `dwm-6.1-1.x86_64.rpm` package, which provides:
```
dwm = 6.0-1.module_1997 + c375c79c
dwm = 6.1-1
dwm (x86-64) = 6.0-1.module_1997 + c375c79c
dwm (x86-64) = 6.1-1
```

The question is, should the files be included?

Reply:
Then it is OK. May I ask you also for one additional test? It is related to https://github.com/rpm-software-management/libdnf/commit/85f1f55afd100afca9fb4e8e36ac5afefaedba14. Try to use a string that will match a summary. And it should not create a reldep.

Reply:
> Try to use a string that will match a summary. And it should not create a reldep.

But this test does it. "dwm*" match the summary too. In fact, it is a rewritten C++ test for Python from the commit you are referencing https://github.com/rpm-software-management/libdnf/commit/85f1f55afd100afca9fb4e8e36ac5afefaedba14.

Reply:
I am sorry, then it is perfect